libs/comparator/tables.js

```diff
@@ -106,7 +113,83 @@ const printDifferences = (apps, appTablesMap) => {
+};
+
+const buildRelationsByIdByTable = tables => {
+    return _.mapValues(_.keyBy(tables, 'name'), table => _.keyBy(table.relations, 'name'));
+};
+
+const buildColumnsById = tables => {
+    return _.keyBy(_.flatMap(tables, 'columns'), 'columnId');
+};
+
+const getColumnByNameAndTable = (columnsByNameByTable, tableName, columnName) => {
+    return columnsByNameByTable[tableName]?.[columnName];
+};
+
+const getTargetRelationByIdAndTable = (relationsByIdByTable, tableName, relationName) => {
+    return relationsByIdByTable[tableName]?.[relationName] || {};
+};
+
+const enrichTargetAppColumnsWithRelationIdentificationNames = (table, targetAppRelationsByIdByTable, targetAppColumnsById) => {
+    const relations = table.relations || [];
+
+    if (relations.length) {
+        relations.forEach(relation => {
+            const { relationIdentificationColumnId } = relation.metaInfo || {};
+
+            if (relationIdentificationColumnId) {
+                const targetRelation = getTargetRelationByIdAndTable(
+                  targetAppRelationsByIdByTable,
+                  relation.fromTableName,
+                  relation.name
+                );
+
+                if (targetRelation?.metaInfo?.relationIdentificationColumnId) {
+                    const relatedColumn = targetAppColumnsById[targetRelation.metaInfo.relationIdentificationColumnId];
+
+                    targetRelation.metaInfo.relationIdentificationColumnName = relatedColumn.name;
+                }
+            }
+        });
+    }
+}
+
+const enrichSchemaColumnsWithRelationIdentificationIds = (table, targetAppColumnsByNameByTable) => {
+    const relations = table.relations || [];
+
+    if (relations.length) {
+        relations.forEach(schemaRelation => {
+            const { relationIdentificationColumnName } = schemaRelation.metaInfo || {};
+
+            if (relationIdentificationColumnName) {
+                const identificationColumn = getColumnByNameAndTable(
+                  targetAppColumnsByNameByTable,
+                  schemaRelation.toTableName,
+                  relationIdentificationColumnName
+                );
+
+                if (identificationColumn) {
+                    schemaRelation.metaInfo.relationIdentificationColumnId = identificationColumn.columnId;
+                }
+            }
+        });
+    }
+}
+
+const enrichColumnsWithRelationIdentificationData = (schema, targetApp) => {
+    const targetAppColumnsByNameByTable = buildColumnsByNameByTable(targetApp.tables);
+    const targetAppRelationsByIdByTable = buildRelationsByIdByTable(targetApp.tables);
+    const targetAppColumnsById = buildColumnsById(targetApp.tables);
+
+    schema.tables.forEach(table => enrichSchemaColumnsWithRelationIdentificationIds(table, targetAppColumnsByNameByTable));
+    targetApp.tables.forEach(table => enrichTargetAppColumnsWithRelationIdentificationNames(table, targetAppRelationsByIdByTable, targetAppColumnsById));
+};
```

Comment:
all this new code should be completely removed from this module

Reply:
@vengrov I removed everything except the logic that pushes `relationIdentificationColumnName` to the `options`